### PR TITLE
test: Add tests for the password checks tooltip

### PR DESCRIPTION
**Pull request by bistaastha**
## Clickup
https://app.clickup.com/t/86cx2t91h

## Code Coverage
Screenshot 2024-11-25 at 1 49 45 PM

## Summary by CodeRabbit

- **New Features**
	- Updated styling for the password check tooltip to enhance visual clarity.
	- Improved password validation logic to handle empty passwords and update criteria dynamically.

- **Tests**
	- Introduced comprehensive test cases for the PasswordCheckTooltipComponent, covering various password validation scenarios and ensuring accurate UI rendering based on validation states.

### CODE REVIEW

File: src/app/shared/components/password-check-tooltip/password-check-tooltip.component.spec.ts [hunk 1]
```diff
@@ -21,4 +22,153 @@ describe('PasswordCheckTooltipComponent', () => {
...
+      component.passwordChecks = {
+        lengthValid: false,
+        uppercaseValid: false,
+        lowercaseValid: false,
+        numberValid: false,
+        specialCharValid: false,
+      };
+      fixture.detectChanges();
+
+      const tooltipText = fixture.debugElement.query(By.css('.tooltip__text')).nativeElement;
```
Comment (arjunaj5) [inline src/app/shared/components/password-check-tooltip/password-check-tooltip.component.spec.ts:118]: we don't have dom helpers in mobile app is it?

**Review by arjunaj5 — APPROVED** [review]

File: src/app/shared/components/password-check-tooltip/password-check-tooltip.component.spec.ts (hunk 1 above)
Comment (bistaastha) [reply src/app/shared/components/password-check-tooltip/password-check-tooltip.component.spec.ts:118]: We have dom helpers, but they weren't used anywhere, will check this

File: src/app/shared/components/password-check-tooltip/password-check-tooltip.component.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<div class="tooltip-container">
```
Comment (Chethan-Fyle) [inline src/app/shared/components/password-check-tooltip/password-check-tooltip.component.html:1]: Why do we have actual component code in the tests PR?

**Review by Chethan-Fyle — COMMENTED** [review]
Why do we have actual component code in the tests PR?

File: src/app/shared/components/password-check-tooltip/password-check-tooltip.component.html (hunk 2 above)
Comment (bistaastha) [reply src/app/shared/components/password-check-tooltip/password-check-tooltip.component.html:1]: I switched branches to master for this one, so the template changes are here now as well. Will switch back to component branch for your reference.

Comment (bistaastha) [reply src/app/shared/components/password-check-tooltip/password-check-tooltip.component.html:1]: Moving to the previous branch is adding a lot of extra changes in the diff

File: src/app/shared/components/password-check-tooltip/password-check-tooltip.component.spec.ts [hunk 3]
```diff
@@ -21,4 +22,153 @@ describe('PasswordCheckTooltipComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('ngOnChanges(): should call validatePassword when ngOnChanges is triggered', () => {
+    spyOn(component, 'validatePassword');
+    component.password = 'ValidPass123!';
+    component.ngOnChanges();
+    expect(component.validatePassword).toHaveBeenCalledTimes(1);
+  });
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/shared/components/password-check-tooltip/password-check-tooltip.component.spec.ts:54]: These are handled in separate test cases

File: src/app/shared/components/password-check-tooltip/password-check-tooltip.component.spec.ts (hunk 1 above)
Comment (bistaastha) [reply src/app/shared/components/password-check-tooltip/password-check-tooltip.component.spec.ts:142]: The DOM helper for this particular method is not present, and the selectors are directly being used

**Pull request merged**